sed/calibrator/energy.py

```diff
@@ -1,5 +1,1900 @@
+                profile, X-direction.
+
+    Returns:
+        float: calculated correction value
+    """
+    if correction_type == "spherical":
+        diameter = kwds.pop("diameter", 50)
+        correction = -(
+            (
+                np.sqrt(
+                    1
+                    + ((x - center[0]) ** 2 + (y - center[1]) ** 2)
+                    / diameter**2,
+                )
+                - 1
+            )
+            * amplitude
+        )
+
+    elif correction_type == "Lorentzian":
+        gamma = kwds.pop("gamma", 700)
+        correction = (
+            100000
+            * amplitude
+            / (gamma * np.pi)
+            * (
+                gamma**2
+                / ((x - center[0]) ** 2 + (y - center[1]) ** 2 + gamma**2)
+                - 1
+            )
+        )
+
+    elif correction_type == "Gaussian":
+        sigma = kwds.pop("sigma", 400)
+        correction = (
+            20000
+            * amplitude
+            / np.sqrt(2 * np.pi * sigma**2)
+            * (
+                np.exp(
+                    -((x - center[0]) ** 2 + (y - center[1]) ** 2)
+                    / (2 * sigma**2),
+                )
+                - 1
+            )
+        )
+
+    elif correction_type == "Lorentzian_asymmetric":
+        gamma = kwds.pop("gamma", 700)
+        gamma2 = kwds.pop("gamma2", gamma)
+        amplitude2 = kwds.pop("amplitude2", amplitude)
+        correction = (
+            100000
+            * amplitude
+            / (gamma * np.pi)
+            * (gamma**2 / ((y - center[1]) ** 2 + gamma**2) - 1)
+        )
+        correction += (
+            100000
+            * amplitude2
+            / (gamma2 * np.pi)
+            * (gamma2**2 / ((x - center[0]) ** 2 + gamma2**2) - 1)
+        )
+
+    else:
         raise NotImplementedError
+
+    return correction
+
+
+def normspec(
+    specs: np.ndarray,
+    smooth: bool = False,
+    span: int = 7,
+    order: int = 1,
+) -> np.ndarray:
+    """
+    Normalize a series of 1D signals.
+
+    Parameters:
+    *specs: list/2D array
+        Collection of 1D signals.
+    smooth: bool | False
+        Option to smooth the signals before normalization.
+    span, order: int, int | 7, 1
+        Smoothing parameters of the LOESS method (see ``scipy.signal.savgol_filter()``).
+
+    **Return**\n
+    normalized_specs: 2D array
+        The matrix assembled from a list of maximum-normalized signals.
+    """
+
+    nspec = len(specs)
+    specnorm = []
+
+    for i in range(nspec):
+
+        spec = specs[i]
+
+        if smooth:
+            spec = savgol_filter(spec, span, order)
+
+        if type(spec) in (list, tuple):
+            nsp = spec / max(spec)
+        else:
+            nsp = spec / spec.max()
+        specnorm.append(nsp)
+
+        # Align 1D spectrum
+        normalized_specs = np.asarray(specnorm)
+
+    return normalized_specs
+
+
+def find_correspondence(
```

Comment:
They all apply to the energy calibration, and are not used anywhere else. But sure, we could put them into a separate module. In mpes, they actually partly were in different files, but I found this quite confusing, because you never knew the dependencies of all the modules. I tried to simplify this.

Reply:
I will keep this as it is for now.